api/v1beta1/module_webhook.go

```diff
@@ -95,19 +95,37 @@ func (m *Module) validateKernelMapping() error {
 
 func (m *Module) validateModprobe() error {
 	modprobe := m.Spec.ModuleLoader.Container.Modprobe
-	moduleNameDefined := modprobe.ModuleName != ""
+	moduleName := modprobe.ModuleName
+	moduleNameDefined := moduleName != ""
 	rawLoadArgsDefined := modprobe.RawArgs != nil && len(modprobe.RawArgs.Load) > 0
 	rawUnloadArgsDefined := modprobe.RawArgs != nil && len(modprobe.RawArgs.Unload) > 0
 
 	if moduleNameDefined {
 		if rawLoadArgsDefined || rawUnloadArgsDefined {
 			return errors.New("rawArgs cannot be set when moduleName is set")
 		}
-		return nil
+	} else if !rawLoadArgsDefined || !rawUnloadArgsDefined {
+		return errors.New("load and unload rawArgs must be set when moduleName is unset")
 	}
 
-	if !rawLoadArgsDefined || !rawUnloadArgsDefined {
-		return errors.New("load and unload rawArgs must be set when moduleName is unset")
+	if modprobe.ModulesLoadingOrder != nil {
+		if len(modprobe.ModulesLoadingOrder) < 2 {
+			return errors.New("if a loading order is defined, at least two values must be defined")
+		}
+
+		if !moduleNameDefined {
+			return errors.New("if a loading order is defined, moduleName must be set")
+		}
+
+		if modprobe.ModulesLoadingOrder[0] != moduleName {
+			return errors.New("if a loading order is defined, the first element must be moduleName")
+		}
+
+		for idx, modName := range modprobe.ModulesLoadingOrder[1:] {
```

Comment:
I think that we need to verify that the whole list is unique, and not moduleName. I mean, that no name can appear in it more then once

Reply:
Absolutely. Now looking for duplicates.